packages/table/src/Table.js

```diff
@@ -59,30 +87,94 @@ class Table extends React.Component {
-            return (
-              <tr key={index} className="row">
-                {selectableRows && this._renderSelectRowButton(index)}
-                {React.Children.map(
-                  children,
-                  column =>
-                    column &&
-                    column.props &&
-                    this._renderColumnCell(row, index, column.props)
-                )}
-              </tr>
-            );
-          })}
-        </tbody>
+        {data.map((row, index) => {
+          const isRowGroup =
+            row[rowGroupKey] && Array.isArray(row[rowGroupKey]);
+          if (isRowGroup) {
+            return this._renderRowGroup(row, index);
+          }
+          return (
+            <tbody key={index} className="body">
+              {this._renderRow(row, index)}
+            </tbody>
+          );
+        })}
       </StyledTable>
     );
   }
 
+  _renderRow = (row, index, subIndex = "", group = false) => {
+    const { selectableRows, collapsed, children } = this.props;
+    const rowKey = `${index},${subIndex}`;
+    const selected = this.state.selectedRows.indexOf(rowKey) !== -1;
+    const grouped = typeof subIndex === "number";
+    const rowProps = { selected, grouped };
+    return (
+      <tr key={rowKey} className="row">
+        {selectableRows && this._renderSelectRowButton(rowKey, rowProps, group)}
+        {React.Children.map(
+          children,
+          column =>
+            column &&
+            column.props &&
+            this._renderColumnCell(row, index, {
+              ...column.props,
+              ...rowProps
+            })
+        )}
+        {collapsed && (
+          <BodyCell className="cell" {...rowProps}>
+            {group && this._renderExpandToggle(index)}
+          </BodyCell>
+        )}
+      </tr>
+    );
+  };
+
+  _renderRowGroup = (row, index) => {
+    const { rowGroupKey } = this.props;
+    const { [rowGroupKey]: childRows, ...parentRow } = row;
+    const expanded =
+      !this.props.collapsed || this.state.expandedRows.indexOf(index) !== -1;
+    return (
+      <tbody className="body" key={index}>
+        {this._renderRow(parentRow, index, "", true)}
+        {expanded &&
+          childRows.map((row, subindex) =>
+            this._renderRow(row, index, subindex)
+          )}
+      </tbody>
+    );
+  };
+
+  _renderExpandToggle = index => {
+    const icon =
+      this.state.expandedRows.indexOf(index) !== -1
+        ? "wg-small-arrow-top"
+        : "wg-small-arrow-bottom";
+    return (
+      <Button icon={icon} onClick={() => this.expandToggleClicked(index)} />
+    );
+  };
+
+  expandToggleClicked = index => {
+    const oldExpandedRows = this.state.expandedRows;
+    const expandedIndexOf = oldExpandedRows.indexOf(index);
+    const expandedRows =
+      expandedIndexOf !== -1
+        ? [
+            ...oldExpandedRows.slice(0, expandedIndexOf),
+            ...oldExpandedRows.slice(expandedIndexOf + 1)
+          ]
```

Comment:
![image](https://media.giphy.com/media/12NUbkX6p4xOO4/giphy.gif)

We could use filter here...
```
const expandedRows = 
  expandedIndexOf !== -1
    ? oldExpandedRows.filter((element, index) => index !== expandedIndexOf)
    : [...oldExpandedRows, index]
```

Reply:
fixed on 01af33dcc8df661329844193494f741e2e6dc4dc